File: CodeForces/1-100/12/A/12A.rb
# Problem 12A: Super Agent
# https://codeforces.com/contest/12/problem/A
# Executed with: ruby 3.3.8 (2025-04-09 revision b200bad6cd) [aarch64-linux-gnu]
# Linux version: Linux kernel 6.17.0-5-generic
# Submitted on: October 26th, 2025
# Codeforces language used: Ruby 3.2.2

str = gets.chomp + gets.chomp + gets.chomp

if str[0] == str[8] && str[1] == str[7] && str[2] == str[6] && str[3] == str[5]
  puts "YES"
else
  puts "NO"
end

File: CodeForces/1-100/12/A/12A.py
# Problem 12A: Super Agent
# https://codeforces.com/contest/12/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: October 24th, 2025
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

input_str = input() + input() + input()

if input_str[0] == input_str[8] and input_str[1] == input_str[7] and input_str[2] == input_str[6] and input_str[3] == input_str[5]:
    print("YES")
else:
    print("NO")
